module/data/item/templates/activities.mjs

```diff
@@ -0,0 +1,71 @@
+import SystemDataModel from "../../abstract.mjs";
+import { ActivitiesField } from "../../fields/activities-field.mjs";
+import UsesField from "../../shared/uses-field.mjs";
+
+/**
+ * Data model template for items with activities.
+ *
+ * @property {ActivityCollection} activities  Activities on this item.
+ * @mixin
+ */
+export default class ActivitiesTemplate extends SystemDataModel {
+
+  /** @override */
+  static LOCALIZATION_PREFIXES = ["DND5E.USES"];
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  static defineSchema() {
+    return {
+      activities: new ActivitiesField(),
+      uses: new UsesField()
+    };
+  }
+
+  /* -------------------------------------------- */
+  /*  Data Migrations                             */
+  /* -------------------------------------------- */
+
+  /**
+   * Migrate the uses data structure from before activities.
+   * @param {object} source  Candidate source data to migrate.
+   */
+  static migrateActivities(source) {
+    ActivitiesTemplate.#migrateUses(source);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Migrate the uses to the new data structure.
+   * @param {object} source  Candidate source data to migrate.
+   */
+  static #migrateUses(source) {
+    if ( foundry.utils.getType(source.uses.recovery) !== "string" ) return;
+
+    // If period is charges, set the recovery type to `formula`
+    if ( source.uses.per === "charges" ) {
+      source.uses.recovery = [{ period: "lr", type: "formula", formula: source.uses.recovery }];
+    }
+
+    // If period is not blank, set recovery type to `recoverAll`
+    else if ( source.uses.per ) {
+      source.uses.recovery = [{ period: source.uses.per, type: "recoverAll" }];
+    }
+
+    // Otherwise, check to see if recharge is set
+    else if ( source.recharge?.value ) {
+      source.uses.recovery = [{ period: "recharge", formula: source.recharge.value }];
+      source.uses.max = 1;
+      source.uses.spent = source.recharge.charged ? 0 : 1;
+      return;
+    }
+
+    else return;
+
+    if ( ("value" in source) && Number.isNumeric(source.uses.max) ) {
+      source.uses.spent = Number(source.uses.max) - source.uses.value;
```

Comment:
What I'm thinking at this point is we do a concrete migration to handle translating `value` to `spent`, and leave the rest of the changes up to in-memory migration. That should allow us to keep the proper available uses for everything in the world where it matters most (since stuff in compendiums will almost always have `0` uses spent).